caps/types.go

```diff
@@ -55,11 +58,26 @@ func (t *Type) Validate(c *Capability) error {
 	if t != c.Type {
 		return fmt.Errorf("capability is not of type %q", t)
 	}
-	// While we don't have any support for type-specific attribute schema,
-	// let's ensure that attributes are totally empty. This will make tests
-	// show that this code is actually being used.
-	if c.Attrs != nil && len(c.Attrs) != 0 {
-		return fmt.Errorf("attributes must be empty for now")
+	// Check that all supported attributes are present
+	for _, attr := range t.SupportedAttrs {
```

Comment:
Hm, if I read this correctly it will fail if a "SupportedAttr" is missing? If thats the case, maybe it should be "t.RequiredAttrs" instead as the name? To make more clear that this is a mandatory attribute?